src/components/internal/useWindowWidth/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react';
+
+export const widths = {
+  xs: 0,
+  sm: 600,
+  md: 960,
+  lg: 1280,
+  xl: 1920,
+} as const;
+
+const getWidth = () => window.innerWidth;
```

Comment:
Use a check like `typeof window === 'undefined'` before using `window.`, so that we're sure that this stuff won't break projects that do server-side rendering (e.g. the Staking app).